Sources/Core/Shared/Operation.swift

```diff
@@ -303,10 +334,14 @@ public class Operation: NSOperation {
      - parameter errors: an `[ErrorType]` defaults to empty array.
      */
     public func cancelWithErrors(errors: [ErrorType] = []) {
-        if !errors.isEmpty {
-            log.warning("Did cancel with errors: \(errors).")
+        stateLock.withCriticalScope {
+            assert(state != .Finishing, "Operation cannot be cancelled after while it is finishing.")
+            assert(state != .Finished, "Operation cannot be cancelled after it has already finished.")
```

Comment:
Was going back and forth over whether this was something that made sense to alert the framework consumer about. In the case of `cancel()`/`cancelWithErrors()`, I agree with your suggestion - it should be something you can safely call at any time, from any thread, without it asserting.
Fixing.